persistence/sql/persister_identity.go

```diff
@@ -565,3 +598,103 @@ func (p *Persister) injectTraitsSchemaURL(ctx context.Context, i *identity.Ident
 	i.SchemaURL = s.SchemaURL(p.r.Config(ctx).SelfPublicURL()).String()
 	return nil
 }
+
+func (p *Persister) getJSONSearchQuery(ctx context.Context, field string, values []string) pop.ScopeFunc {
+	return func(q *pop.Query) *pop.Query {
+		// Removing this would allow SQL Injection. Never remove this.
+		if !validFields.MatchString(field) {
+			return q
+		}
+
+		prefix, key := extractFieldAndInnerFields(field)
+
+		switch p.Connection(ctx).Dialect.Name() {
+		case "sqlite3", "mysql", "mariadb":
+			for _, value := range values {
+				if key == "" {
+					q = q.Where(fmt.Sprintf(`json_extract(%s, '$') = ?`, prefix), value)
```

Comment:
The `%s` should be escaped, something like:

```suggestion
					q = q.Where(fmt.Sprintf(`json_extract(`%s`, '$') = ?`, prefix), value)
```